Lib/fontTools/ttLib/tables/O_S_2f_2.py

```diff
@@ -475,22 +476,19 @@ def recalcUnicodeRanges(self, ttFont, pruneOnly=False):
 )
 
 
-_unicodeRangeSets = []
+_unicodeStarts = []
+_unicodeValues = [None]
 
-def _getUnicodeRangeSets():
-	# build the sets of codepoints for each unicode range bit, and cache result
-	if not _unicodeRangeSets:
-		for bit, blocks in enumerate(OS2_UNICODE_RANGES):
-			rangeset = set()
-			for _, (start, stop) in blocks:
-				rangeset.update(set(range(start, stop+1)))
-			if bit == 57:
-				# The spec says that bit 57 ("Non Plane 0") implies that there's
-				# at least one codepoint beyond the BMP; so I also include all
-				# the non-BMP codepoints here
-				rangeset.update(set(range(0x10000, 0x110000)))
-			_unicodeRangeSets.append(rangeset)
-	return _unicodeRangeSets
+def _getUnicodeRanges():
+	# build the ranges of codepoints for each unicode range bit, and cache result
+	if not _unicodeStarts:
+		unicodeRanges = {
```

Comment:
A list of tuples would be good enough. Using a dict only makes it more complicated, and possibly slower. A list can be sorted in-place, unlike a dict.

Reply:
```python
unicodeRanges = [
    (start, stop, bit) for bit, blocks in enumerate(OS2_UNICODE_RANGES)
    for _, (start, stop) in val]
# The spec says ... etc.
unicodeRanges.append((0x10000, 0x10FFFF, 57))
unicodeRanges.sort()
for start, stop, bit in unicodeRanges:
    ... etc.
```

Reply:
> A list of tuples would be good enough. Using a dict only makes it more complicated, and possibly slower. A list can be sorted in-place, unlike a dict.

You’re right, it’s faster.

> ```python
> unicodeRanges.append((0x10000, 0x10FFFF, 57))
> ```

I don’t think it’s possible, because `(0x10000, 0x10FFFF)` overlaps with other ranges. Did I miss something?

Reply:
> I don’t think it’s possible, because (0x10000, 0x10FFFF) overlaps other ranges. Did I miss something?

Oops, I think you're right, sorry.

Reply:
> A list of tuples would be good enough

Right. @liZe will you change that?

Reply:
> Right. @liZe will you change that?

Done!